### PR TITLE
NXBT-2161 retrieve archived artifacts from upstream builds

### PR DESCRIPTION
**Pull request by nxmatic**

### CODE REVIEW

File: vars/rawBuild.groovy [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * Navigates the upstream run of the current cuild
+ **/
+@NonCPS
+def upstreamRun() {
+    def upstreamCause = upstreamCause()
+    if (upstreamCause == null) {
+        return null
+    }
+    return upstreamCause.upstreamRun
+}
```
Comment (jcarsique) [inline vars/rawBuild.groovy]: isn't it overkill? this looks like equivalent to Groovy Elvis operator: `upstreamCause?.upstreamRun`

File: vars/rawBuild.groovy [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Copy filtered upstream artifacts to target dir
+ **/
+@NonCPS
+def copyUpstreamArtifacts(String filename, String target) {
+    def upstreamRun = upstreamRun()
+    if (upstreamRun == null) {
+        return false
+    }
+    copyArtifacts(projectName: upstreamRun.parent.fullName, selector: [$class: 'TriggeredBuildSelector'], filter: '**/'+filename, target: target, flatten: true, optional: false)
```
Comment (jcarsique) [inline vars/rawBuild.groovy]: you may consider using a default argument for `flatten: true` as I guess it is often used with flatten=true
http://groovy-lang.org/objectorientation.html#_default_arguments

**Review by jcarsique — APPROVED** [review]
:+1: (except `upstreamRun()` if confirmed as useless)

**Comment by pgmillon** [comment]
👍

**Pull request merged**